fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -872,6 +860,20 @@ public FHIRRestOperationResponse doPatchOrUpdatePersist(FHIRPersistenceEvent eve
         }
     }
 
+    private void handleIfNoneMatchExisted(String type, String id, FHIRRestOperationResponse ior,
+            Integer ifNoneMatchVersion) throws FHIRPersistenceIfNoneMatchException {
+        ior.setResource(null); // the resource shouldn't be needed for either case (302 or 412)
```

Comment:
I think you mean 304. I guess this works now because we're using the ResourceResult to help build the response.

Reply:
oops, yes, 304
```suggestion
        ior.setResource(null); // the resource shouldn't be needed for either case (304 or 412)
```